code/pages/DMSDocumentCartCheckoutPage.php

```diff
@@ -0,0 +1,258 @@
+            )->setValue('NZ')
+        );
+
+        $form = DMSDocumentRequestForm::create($this, 'DMSDocumentRequestForm', $fields, FieldList::create());
+        if ($receiverInfo = $this->getCart()->getReceiverInfo()) {
+            $form->loadDataFrom($receiverInfo);
+        }
+
+        if ($message = $this->request->getVar('message')) {
+            $form->sessionMessage($this->ThanksMessage, 'good');
+        } else {
+            $form->clearMessage();
+        }
+        $this->extend('updateDMSDocumentRequestForm', $form);
+
+        return $form;
+    }
+
+    /**
+     * Sends an email to both the configured recipient as well as the requester. The
+     * configured recipient is bcc'ed to the email in order to fulfill it.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateSend($email)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @return mixed
+     *
+     * @throws DMSDocumentCartException
+     */
+    public function send()
+    {
+        $member = $this->CartEmailRecipient();
+        if ($member->exists()) {
+            $cart = $this->getCart();
+            $from = Config::inst()->get('Email', 'admin_email');
+            $emailAddress = ($info = $cart->getReceiverInfo()) ? $info['ReceiverEmail'] : $from;
+            $email = Email::create(
+                $from,
+                $emailAddress,
+                _t('DMSDocumentCartCheckoutPage.EMAIL_SUBJECT', 'Request for Printed Publications')
+            );
+            $email->setBcc($member->Email);
+            $renderedCart = $cart->renderWith('DocumentCart_email');
+            $body = _t(
+                'DMSDocumentCartCheckoutPage.EMAIL_BODY',
+                '<p>A request for printed publications has been submitted with the following details:</p>'
+            );
+            $body .= $renderedCart->getValue();
+            $email->setBody($body)->setReplyTo($emailAddress);
+            $this->extend('updateSend', $email);
+            return $email->send();
+        } else {
+            throw new DMSDocumentCartException('No recipient has been configured. Please do so from the CMS');
+        }
+    }
+
+    /**
+     * Handles form submission.
+     * Totals requested are updated, delivery details added, email sent for fulfillment
+     * and print request totals updated.
+     *
+     * @param array          $data
+     * @param Form           $form
+     * @param SS_HTTPRequest $request
+     *
+     * @return SS_HTTPResponse
+     */
+    public function doRequestSend($data, Form $form, SS_HTTPRequest $request)
+    {
+        $this->updateCartItems($data);
+        $this->updateCartReceiverInfo($data);
+        $this->send();
+        $this->trackTimestampedPrintRequest();
+        $this->getCart()->saveSubmission($form);
+        $this->getCart()->emptyCart();
+        return $this->redirect("{$this->Link()}?message=1");
```

Comment:
As mentioned previously, can we get rid of this query param and just use an action like `/checkout/complete`